### PR TITLE
DB\Quest: Mage Summoner

### PR DESCRIPTION
**Pull request by poszer**
##### CHANGES PROPOSED:

-  Added gossip menu option for Shaman Trainer
-  Updated faction for creature Minor Water Guardian
-  Updated flags for creature Sarilus Foulborne
-  Added SAI for Creature Sarilus Foulborne

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

Tested in game, the quest is working fine now and can be done.

##### HOW TO TEST THE CHANGES:

1. For missing Shaman Trainer ( .go creature 4664 ) and speak with NPC
2. There should be '' Train '' option.
3. For Quest Mage Summoner, Use Alliance character and ( .quest add 1017 )
4. .go creature 15084 and fight it 
5. The Creature should cast Spells and Summons Guardians

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1548144375618827200.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@ENTRY, 0, 0, 0, 0, 0, 100, 0, 4000, 4000, 5000, 8000, 11, 20806, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, "When in combat and timer at the begining between 4000 and 4000 ms (and later repeats every 5000 and 8000 ms) - Self: Cast spell Frostbolt (20806) on Random hostile // "),
+(@ENTRY, 0, 1, 0, 0, 0, 100, 1, 0, 1000, 0, 0, 11, 6490, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "When in combat and timer at the begining between 0 and 1000 ms (and later repeats every 0 and 0 ms) - Self: Cast spell Sarilus's Elementals (6490) on Self // "),
+(@ENTRY, 0, 2, 0, 7, 0, 100, 0, 0, 0, 0, 0, 41, 1000, 0, 0, 0, 0, 0, 9, 3950, 0, 40, 0, 0, 0, 0, "On evade - Creature Minor Water Guardian (3950) in 0 - 40 yards: Despawn in 1000 ms // ");
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1548144375618827200.sql]: why all the "//" at the end? :P

Comment (poszer) [reply data/sql/updates/pending_db_world/rev_1548144375618827200.sql]: Edited and removed :tada:

**Comment by wetbrownsauce** [comment]
Tested. 

- Sagorne Creststrider now correctly has the text "Teach me the ways of the spirits" instead of a blank gossip option. All his gossip options open the correct corresponding features. 

- Sarilus Foulborne summons water elementals on pull, then proceeds to use frostbolt in combat.
*note: Sarilus will always run to the player to be in melee range, even though he casts frostbolt every couple seconds. Maybe he should stay at casting range until OOM?

**Comment by poszer** [comment]
> Tested.
> 
> * Sagorne Creststrider now correctly has the text "Teach me the ways of the spirits" instead of a blank gossip option. All his gossip options open the correct corresponding features.
> * Sarilus Foulborne summons water elementals on pull, then proceeds to use frostbolt in combat.
>   *note: Sarilus will always run to the player to be in melee range, even though he casts frostbolt every couple seconds. Maybe he should stay at casting range until OOM?

Thanks for testings. But, Yeah, there is also another quest with this creature for Druids, where this mage moving and just attacking random npc spawned ( quest npc ) without casting Frostbolt, like in this quest case. I Will find name of quest so you can take a look on it also

**Comment by BarbzYHOOL** [comment]
@poszer @wetbrownsauce  thank you AC staff!

**Pull request merged**